refactor: remove unused methods in classes `CDBIterator,CDBWrapper,CCoinsViewDBCursor`

theStack opened the pull request:
The `GetValueSize` methods haven't been used since the chainstate db cache has been switched from per-tx to per-txout model years ago (PR #10195, commit d342424301013ec47dc146a4beb49d5c9319d80a). The `CompactRange` is unused since the txindex migration code was removed (PR https://github.com/bitcoin/bitcoin/pull/22626, commit https://github.com/bitcoin/bitcoin/commit/fa20f815a9cb438c5ab61e97a453612ddd8b21b5).

theStack committed:
refactor: remove unused methods `{CDBIterator,CCoinsViewDBCursor}::GetValueSize()`

These methods haven't been used since the chainstate db cache has been
switched from per-tx to per-txout model years ago (PR #10195, commit
d342424301013ec47dc146a4beb49d5c9319d80a).

w0xlt reviewed (approved):
Code Review ACK https://github.com/bitcoin/bitcoin/pull/25438/commits/fb38c6e21f064e23b63a46d15adb873029463cff

laanwj commented:
Code review ACK fb38c6e21f064e23b63a46d15adb873029463cff

furszy reviewed (approved):
ACK fb38c6e2

laanwj commented:
> Btw, checking at this dbwrapper.h file, isn't CompactRange unused as well?

Yes. It seems like it!
A direct call to leveldb's `CompactRange` is used in `CDBWrapper::CDBWrapper` in `-forcecompactdb` handling. But our own function is never used.

theStack committed:
refactor: remove unused method `CDBWrapper::CompactRange`

This method hasn't been used since the txindex migration code has been
removed (PR #22626, commit fa20f815a9cb438c5ab61e97a453612ddd8b21b5).

Co-authored-by: furszy <matiasfurszyfer@protonmail.com>

fanquake reviewed (approved):
ACK e4b4db561049c97c956e5b856713dcf63ac3e2f0

furszy reviewed (approved):
re-ACK e4b4db56

MarcoFalke commented:
`Clear` may be unused as well? https://marcofalke.github.io/btc_cov/total.coverage/src/dbwrapper.h.gcov.html#66

theStack commented:
> `Clear` may be unused as well? https://marcofalke.github.io/btc_cov/total.coverage/src/dbwrapper.h.gcov.html#66

Seems like it is used in `CCoinsViewDB::BatchWrite` (https://github.com/bitcoin/bitcoin/blob/01e9e2d1ca7fc08d65663b398c71ecec6a01f686/src/txdb.cpp#L154):
```
txdb.cpp:154:19: error: no member named 'Clear' in 'CDBBatch'
            batch.Clear();
            ~~~~~ ^
1 error generated.
gmake[2]: *** [Makefile:10066: libbitcoin_node_a-txdb.o] Error 1
```

laanwj reviewed (approved):
Code review ACK e4b4db561049c97c956e5b856713dcf63ac3e2f0
